pkg/module/drift/config.go

```diff
@@ -422,6 +422,13 @@ func (d *Drift) applyProvider(cfg *ProviderConfig, p *cqproto.GetProviderSchemaR
 
 	if len(cfg.versionConstraints) > 0 {
 		pver, err := version.NewSemver(p.Version)
+		if err == nil {
```

Comment:
what is this? feels not really part of this PRs scope

Reply:
drift version checks don't work any more due to `-SNAPSHOT-commithash` being added, not sure why. this addresses the case, but I can split it later.